02_cancer_type_classification/nbconverted/lasso_range_all.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ### LASSO parameter range experiments, summary across all genes
+
+# In[1]:
+
+
+import os
+import itertools as it
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr
+import seaborn as sns
+
+import pancancer_evaluation.config as cfg
+import pancancer_evaluation.utilities.data_utilities as du
+import pancancer_evaluation.utilities.analysis_utilities as au
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# In[2]:
+
+
+base_results_dir = os.path.join(
+    cfg.repo_root, '02_cancer_type_classification', 'results', 'lasso_range_valid'
+)
+
+training_dataset = 'all_other_cancers'
+results_dir = os.path.join(base_results_dir, training_dataset)
+
+metric = 'aupr'
+
+output_plots = True
+output_plots_dir = cfg.cancer_type_lasso_range_dir
+
+
+# ### Get coefficient information for each lasso penalty
+
+# In[3]:
+
+
+nz_coefs_df = []
+
+# get pancancer coefs info for now
+for coef_info in au.generate_nonzero_coefficients_lasso_range(results_dir):
+    (gene,
+     cancer_type,
+     seed,
+     lasso_param,
+     coefs_list) = coef_info
+    for fold_no, coefs in enumerate(coefs_list):
+        nz_coefs_df.append(
+            [gene, cancer_type, lasso_param, seed, fold_no, len(coefs)]
+        )
+        
+nz_coefs_df = pd.DataFrame(
+    nz_coefs_df,
+    columns=['gene', 'cancer_type', 'lasso_param', 'seed', 'fold', 'nz_coefs']
+)
+nz_coefs_df.head()
+
+
+# ### Get performance information for each lasso penalty
+
+# In[4]:
+
+
+perf_df = au.load_prediction_results_lasso_range(results_dir, training_dataset)
+perf_df.head()
+
+
+# ### Compare feature selection with performance
+
+# In[5]:
+
+
+coefs_perf_df = (nz_coefs_df
+    .rename(columns={'cancer_type': 'holdout_cancer_type'})
+    .merge(perf_df[perf_df.signal == 'signal'],
+           on=['gene', 'holdout_cancer_type', 'seed', 'fold', 'lasso_param'])
+    .drop(columns=['signal'])
+)
+
+coefs_perf_df.head()
+
+
+# In[6]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+
+sns.histplot(coefs_perf_df.nz_coefs, log_scale=(True, False))
+plt.gca().axvline(coefs_perf_df.nz_coefs.quantile(q=0.1), linestyle='--')
+plt.title('Distribution of feature count across cancer types/folds')
+plt.xlabel('Number of nonzero features')
+
+coefs_perf_df.loc[coefs_perf_df.nz_coefs.sort_values()[:8].index, :]
+
+
+# In[7]:
+
+
+# look at correlation for each cancer type individually
+# positive correlation => more features, better performance
+corr_cancer_type_df = []
+nz_coefs_cutoff = coefs_perf_df.nz_coefs.quantile(q=0.1)
+
+if nz_coefs_cutoff is not None:
```

Comment:
This is a confusing line to read, could you throw a comment on top to show what this line is doing?